bubble_sort.rb

```diff
@@ -0,0 +1,42 @@
+def bubble_sort(sample_array)
+  array_length = sample_array.length
+  i = 0
+  while i < (array_length - 1)
+    j = 0
+    while j < (array_length - 1) - i
+      if sample_array[j] > sample_array[j + 1]
+        temp = sample_array[j]
+        sample_array[j] = sample_array[j + 1]
+        sample_array[j + 1] = temp
+      end
+      j += 1
+    end
+    i += 1
+  end
+  p sample_array
+end
+
+# Test code
+some_array = [8, 9, 4, 7, 12, 6, 2, 3, 5]
+bubble_sort(some_array)
+
+def bubble_sort_by(sample_array)
+  array_length = sample_array.length
+  i = 0
+  while i < array_length - 1
+    j = 0
+    while j < (array_length - 1) - i
+      if (yield sample_array[j], sample_array[j + 1]).positive?
+        temp = sample_array[j]
+        sample_array[j] = sample_array[j + 1]
+        sample_array[j + 1] = temp
+      end
+      j += 1
+    end
+    i += 1
+  end
+  puts sample_array
+end
```

Comment:
- [ ] bubble_sort_by is failing when given an array of Integers let's fix it 

![image](https://user-images.githubusercontent.com/52209067/89344785-aa91a700-d663-11ea-88a1-b61841e72396.png)